storage/scylladb_identity.c

```diff
@@ -33,25 +33,32 @@ void db_identity_free(db_identity_t** obj) {
   *obj = NULL;
 }
 
-status_t db_set_identity_id(db_identity_t* obj, cass_int64_t id) {
+status_t db_set_identity_uuid(db_identity_t* obj, CassUuid* uuid) {
   if (obj == NULL) {
     ta_log_error("NULL pointer to ScyllaDB identity object\n");
     return SC_TA_NULL;
   }
-  if (id <= 0) {
-    ta_log_error("Invaild ID\n");
-    return SC_STORAGE_INVAILD_INPUT;
+  if (uuid == NULL) {
```

Comment:
`uuid` is of type `CassUuid`. Is proper to compare with `NULL`, which is defined as `(void*) 0`?

Reply:
It is declared as a pointer to `CassUuid` here.
`status_t db_set_identity_uuid(db_identity_t* obj, CassUuid* uuid) `

Reply:
Shall we prevent the condition `!uuid || !*uuid`?

Reply:
Do `!uuid` and `uuid == NULL` both aim to check the condition that `uuid` is equal to `(void*)0`?

Since CassUuid is the struct with the following definition.
```
typedef struct CassUuid_ {
  cass_uint64_t time_and_version;
  cass_uint64_t clock_seq_and_node;
} CassUuid;
```
Is `!*uuid` not zero when all members in it are 0? For example {0,0} here.

Reply:
`!*uuid` is meant to validate the range of UUID value distributions.

Reply:
In `time_and_version`, the most significant 4 bits represent the version, I could use them to check we get the right version of UUID.
Other bits in `time_and_version` and `clock_seq_and_node` are randomly generated.